pjmedia/src/pjmedia/sdp_neg.c

```diff
@@ -447,6 +447,17 @@ PJ_DEF(pj_status_t) pjmedia_sdp_neg_send_local_offer( pj_pool_t *pool,
 	neg->state = PJMEDIA_SDP_NEG_STATE_LOCAL_OFFER;
 	neg->neg_local_sdp = pjmedia_sdp_session_clone(pool, 
 						       neg->active_local_sdp);
+
+#if PJMEDIA_SDP_NEG_COMPARE_BEFORE_INC_VERSION
+    	if (pjmedia_sdp_session_cmp(neg->active_local_sdp, 
+    				    neg->initial_sdp, 0) != PJ_SUCCESS)
+    	{
+	    neg->active_local_sdp->origin.version++;
+    	}    
+#else
+    	neg->active_local_sdp->origin.version++;
+#endif
+
 	*offer = neg->active_local_sdp;
```

Comment:
Hmm.. this is a different but related issue. Just realized that the offer output is the active_local_sdp, while it perhaps should be neg_local_sdp, note that the future sdp nego will only use neg_local_sdp instead of active_local_sdp. So far this works fine because they are identic, but now they are no longer identic as the version is different (so perhaps the future active local SDP, nego result, will have the wrong version? anyway, theoretically returning active_local_sdp here is wrong).

In other hands, not sure if changing the output offer to neg_local_sdp is safe, need some tests (e.g: GitHub CI).

Reply:
Local unit tests are okay here when returning `neg_local_sdp`.

Also, what if here we get rid of the check `PJMEDIA_SDP_NEG_COMPARE_BEFORE_INC_VERSION` and always compare the SDP? This is because in my understanding, one of the purpose of introducing the setting is for backward compatibility and that's why the default is zero (i.e. previously we always increase the version). But here is the other way around, we currently do not increase the version and the change will do it.

Reply:
Always increasing the version is safe (standard wise) and faster (for local endpoint), hence we use it as default. But then some apps want to increase only when something really changes (so remote won't need to renegotiate SDP & re-init media due to different SDP version, when actually nothing changes), this is the purpose of the macro.

Here `pjmedia_sdp_neg_send_local_offer()` can always increase the version (without check for changes), it is safe & faster. But then ... as described above. So the macro check purpose still applies here I guess?